bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/LedgerCommand.java

```diff
@@ -0,0 +1,203 @@
+import java.io.IOException;
+import java.util.function.Function;
+
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.bookkeeper.bookie.Bookie;
+import org.apache.bookkeeper.bookie.InterleavedLedgerStorage;
+import org.apache.bookkeeper.bookie.LedgerCache;
+import org.apache.bookkeeper.bookie.LedgerEntryPage;
+import org.apache.bookkeeper.bookie.SortedLedgerStorage;
+import org.apache.bookkeeper.bookie.storage.ldb.DbLedgerStorage;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.tools.cli.helpers.BookieCommand;
+import org.apache.bookkeeper.tools.framework.CliFlags;
+import org.apache.bookkeeper.tools.framework.CliSpec;
+import org.apache.bookkeeper.util.LedgerIdFormatter;
+import org.apache.commons.lang.mutable.MutableLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Command to dump ledger index entries into readable format.
+ */
+public class LedgerCommand extends BookieCommand<LedgerCommand.LedgerFlags> {
+
+    static final Logger LOG = LoggerFactory.getLogger(LedgerCommand.class);
+
+    private static final String NAME = "ledger";
+    private static final String DESC = "Dump ledger index entries into readable format";
+
+    private LedgerIdFormatter ledgerIdFormatter;
+
+    private Function<String, Boolean> print = this::printInfoLine;
+
+    public void setPrint(Function<String, Boolean> print) {
+        this.print = print;
+    }
+
+    public LedgerCommand() {
+        this(new LedgerFlags());
+    }
+
+    public LedgerCommand(LedgerIdFormatter ledgerIdFormatter) {
+        this(new LedgerFlags());
+        this.ledgerIdFormatter = ledgerIdFormatter;
+    }
+
+    private LedgerCommand(LedgerFlags flags) {
+        super(CliSpec.<LedgerFlags>newBuilder().withName(NAME).withDescription(DESC).withFlags(flags).build());
+    }
+
+    /**
+     * Flags for ledger command.
+     */
+    @Accessors(fluent = true)
+    @Setter
+    public static class LedgerFlags extends CliFlags {
+
+        @Parameter(names = { "-id", "--ledgerId" }, description = "Specific ledger id", required = true)
+        private long ledgerId;
+
+        @Parameter(names = { "-m", "--meta" }, description = "Print meta information")
+        private boolean meta;
+
+        @Parameter(names = { "-l", "--ledgeridformatter" }, description = "Set ledger id formatter")
+        private String ledgerIdFormatter = "";
+    }
+
+    @Override
+    public boolean apply(ServerConfiguration conf, LedgerFlags cmdFlags) {
+        initLedgerIdFormatter(conf, cmdFlags);
+        long ledgerId = cmdFlags.ledgerId;
+        if (conf.getLedgerStorageClass().equals(DbLedgerStorage.class.getName())) {
+            // dump ledger info
+            if (!dumpLedgerInfo(ledgerId, conf)) {
+                return false;
+            }
+        } else if (conf.getLedgerStorageClass().equals(SortedLedgerStorage.class.getName()) || conf
```

Comment:
```suggestion
        } else if (conf.getLedgerStorageClass().equals(SortedLedgerStorage.class.getName()) 
            || conf.getLedgerStorageClass().equals(InterleavedLedgerStorage.class.getName())) {
```

Reply:
That 's because style check. :) I will fix it.